### PR TITLE
fix English Doc of API:layers.py_func/sum

### PR DESCRIPTION
**Pull request by zhwesky2010**
fix English Doc of 2 API: py_func、sum.

**1.py_func** : This PR is the same with chinese [PR1227](https://github.com/PaddlePaddle/FluidDoc/pull/1227)

MacHi 2019-10-10 20-21-56
MacHi 2019-10-10 20-22-19

**2.sum**: This PR is the same with chinese [PR1376](https://github.com/PaddlePaddle/FluidDoc/pull/1376)

MacHi 2019-10-10 20-24-19
MacHi 2019-10-10 20-24-46

### CODE REVIEW

**Review by zhaoyuchen2018 — DISMISSED** [review]
SUM LGTM

File: paddle/fluid/operators/sum_op.cc [hunk 1]
```diff
@@ -157,19 +157,22 @@ class SumOp : public framework::OperatorWithKernel {
 class SumOpMaker : public framework::OpProtoAndCheckerMaker {
  public:
   void Make() override {
-    AddInput("X", "(vector<Tensor>) The input tensors of sum operator.")
+    AddInput("X",
+             "One or more Variable. If more than one Variable are input, "
```
Comment (liupluswei) [inline paddle/fluid/operators/sum_op.cc]: A varaible list. The shape and data type of the list elements should be consistent.

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -11866,21 +11866,70 @@ def gaussian_random_batch_size_like(input,
...
+            1570701754	the sum of input0 and input1: 	The place is:CPUPlace
+            Tensor[sum_0.tmp_0]
+                shape: [2,3,]
+                dtype: l
+                data: 8,8,8,8,8,8,
+
+            #the sum of input0 and input1 is 2-D Tensor with shape [2,3].
+            #dtype is the corresponding C++ data type, which may vary in different environments, 
+            #but the essence is the same.
+            #Eg: if the data type of tensor is int64, then the corresponding C++ data type is int64_t,
```
Comment (liupluswei) [inline python/paddle/fluid/layers/nn.py]: leava a space between '#' and the first word:
'#' the sum ...
'#' dtype is ...

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -11866,21 +11866,70 @@ def gaussian_random_batch_size_like(input,
...
+            #The printed result is:
+            1570701754	the sum of input0 and input1: 	The place is:CPUPlace
+            Tensor[sum_0.tmp_0]
+                shape: [2,3,]
+                dtype: l
+                data: 8,8,8,8,8,8,
+
+            #the sum of input0 and input1 is 2-D Tensor with shape [2,3].
+            #dtype is the corresponding C++ data type, which may vary in different environments, 
+            #but the essence is the same.
```
Comment (liupluswei) [inline python/paddle/fluid/layers/nn.py]: 最好去掉这句

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -14253,85 +14302,90 @@ def __call__(self, *args):
...
 def py_func(func, x, out, backward_func=None, skip_vars_in_backward_input=None):
     """
-    PyFunc Operator.
-
-    User can use :code:`py_func` to register operators in Python side.
-    The inputs of :code:`func` is :code:`LoDTensor` and outputs can be
-    numpy array or :code:`LoDTensor`. Paddle would call the registered
-    :code:`func` in forward part, and call :code:`backward_func` in
-    backward part (if :code:`backward_func` is not None).
+    PaddlePaddle Fluid registers the Python OP through this API. The forward
```
Comment (liupluswei) [inline python/paddle/fluid/layers/nn.py]: This API is used to register customized OP to Fluid.

File: python/paddle/fluid/layers/nn.py [hunk 5]
```diff
@@ -14253,85 +14302,90 @@ def __call__(self, *args):
...
+    is the output of ``func``, whose type can be either LoDTensor or NumPy array.
 
-    User should set the right data type and shape of :code:`out` before
-    calling this function. However, data types and shapes of gradients of
-    :code:`out` and :code:`x` would be inferred automatically.
+    The input of the backward function ``backward_func`` is ``x``, ``out`` and 
+    the gradient of ``out``. If some variables of ``out`` have no gradient, the 
+    relevant input variable of ``backward_func`` is None. If some variables of 
+    ``x`` do not have a gradient, the user should return None in ``backward_func``
+    voluntarily.
```
Comment (liupluswei) [inline python/paddle/fluid/layers/nn.py]: 去掉voluntarily？

File: python/paddle/fluid/layers/nn.py [hunk 6]
```diff
@@ -14253,85 +14302,90 @@ def __call__(self, *args):
...
-    no gradient, users should return None.
+    The data type and shape of ``out`` should also be set correctly before this 
+    API is called, and the data type and shape of the gradient of ``out`` and 
+    ``x`` will be inferred automatically.
 
-    This function can also be used to debug the running network. User can
-    add a :code:`py_func` operator without output, and print input
-    :code:`x` inside :code:`func`.
+    This API can also be used to debug a running network by setting the ``py_func``
+    to have no output and printing the input ``x`` in ``func``.
```
Comment (liupluswei) [inline python/paddle/fluid/layers/nn.py]: This API can also be used to debug the neural network by passing a function that only print variables to ``py_func``

File: python/paddle/fluid/layers/nn.py [hunk 7]
```diff
@@ -14253,85 +14302,90 @@ def __call__(self, *args):
...
+            Variable | tuple[Variable] | list[Variale], in which Variable is LoDTensor.
+        out (Variable): the output of the forward function ``func``, its type can be
+            Variable | tuple[Variable] | list[Variale], in which Variable can be either 
+            LoDTensor or NumPy array. Since Paddle cannot automatically infer the shape
+            and data type of ``out``, ``out`` must be created in advance.
+        backward_func (callable, optional): the backward function of the registered Python OP. Its
+            default value is None, which means there is no reverse calculation. If it is not None, 
+            ``backward_func`` is called to calculate the gradient of ``x`` when the network is
+            at backwward runtime.
+        skip_vars_in_backward_input (Variable, optional): Unnecessary variables in ``backward_func``, it
```
Comment (liupluswei) [inline python/paddle/fluid/layers/nn.py]: Unnecessary variables in ``backward_func``  有些没讲清楚，unnecessary太主观了，你得告诉用户什么时候需要设置这个
比如used to limit the input variable list of ``backward_func``

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/nn.py]: 已去掉

File: python/paddle/fluid/layers/nn.py (hunk 4 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 5 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 6 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/nn.py]: Done

**Review by liupluswei — DISMISSED** [review]
LGTM

**Review by zhaoyuchen2018 — APPROVED** [review]
sum LGTM

**Review by lanxianghit — APPROVED** [review]

**Review by wanghaoshuang — APPROVED** [review]

**Pull request merged**